### PR TITLE
Readme and license

### PR DESCRIPTION
**Pull request by adthrasher**
Updates to the README to include a quickstart as well as more detailed instructions for using either Docker or local. Added Apache 2.0 license. Added documentation on how to verify removal of host variants in the sample data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,10 +179,10 @@ container, not the host, e.g.,
...
   class: File
   path: /data/sample.bam
-ref_db_prefix: /references/ref.fa
+ref_db_prefix: /reference/ref.fa
 ```
 
-The following is an example `run` command where files are stored in `test/{data,references}`. Outputs are saved in `test/results`.
+The following is an example `run` command where files are stored in `test/{data,reference}`. Outputs are saved in `test/results`.
 
 This example assumes you are running against Mus Musculus (genome build MGSCv37). Set the path to the folder containing your reference data
```
Comment (zaeleus) [inline README.md]:
```suggestion
This example assumes you are running against Mus musculus (genome build MGSCv37). Set the path to the folder containing your reference data
```

File: README.md [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
 
-	git clone https://github.com/stjude/XenoCP.git
-	cd XenoCP
-	gradle :xenocp:installDist
+XenoCP can be run in the cloud on DNAnexus at
+https://platform.dnanexus.com/app/stjude_xenocp
 
-Download contaminant genomic reference and update `sample_data/input_data/inputs-local.yml` with the path to the reference data.
+The easiest way to get XenoCP running locally is using Docker, as `docker build`
+will install all of the dependencies into the container:
```
Comment (zaeleus) [inline README.md]:
```suggestion
creates an image with all of the dependencies:
```

File: README.md [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
+	  /data/inputs.yml
 	
 ## Introduction to XenoCP
 
-XenoCP is a cloud-based tool for cleansing mouse reads in xenograft BAMs. XenoCP can be easily incorporated into any workflow as it takes a BAM file
-as input and efficiently cleans up the mouse contamination and gives a clean human BAM output that could be used for downstream
-genomic analysis. 
-
-St. Jude cloud version: ?
+XenoCP takes a BAM with xenograft reads mapped to the graft genome (e.g. human).
```
Comment (zaeleus) [inline README.md]:
```suggestion
XenoCP takes a BAM with xenograft reads mapped to the graft genome (e.g., human).
```

File: README.md [hunk 4]
```diff
@@ -1,23 +1,39 @@
...
+It extracts aligned reads and remaps to the host genome (e.g. mouse) to
```
Comment (zaeleus) [inline README.md]:
```suggestion
It extracts aligned reads and remaps to the host genome (e.g., mouse) to
```

File: README.md [hunk 5]
```diff
@@ -1,23 +1,39 @@
...
+determine whether the reads are from host or graft.  Output is a copy of the
```
Comment (zaeleus) [inline README.md]:
```suggestion
determine whether the reads are from the host or graft.  The output is a copy of the
```

File: README.md [hunk 6]
```diff
@@ -1,23 +1,39 @@
...
+original BAM with host reads marked as unmapped.
```
Comment (zaeleus) [inline README.md]:
```suggestion
original BAM with host reads are marked as unmapped.
```

File: README.md [hunk 7]
```diff
@@ -55,30 +71,28 @@ disabled.
...
+Once the prerequisites are satisfied, build XenoCP using Gradle. 
 
 ```
-$ mkdir results
-$ cwltool --outdir results cwl/xenocp.cwl sample_data/input_data/inputs_local.yml
+$ gradle installDist
 ```
 
-[CWL]: https://www.commonwl.org/
+Add the artifacts under `build/install/xenocp/lib` to your Java CLASSPATH.
```
Comment (zaeleus) [inline README.md]:
```suggestion
Add the artifacts under `build/install/xenocp/lib` to your Java `CLASSPATH`.
```

File: README.md [hunk 8]
```diff
@@ -55,30 +71,28 @@ disabled.
...
+Add the artifacts under `build/install/xenocp/bin` to your PATH.
```
Comment (zaeleus) [inline README.md]:
```suggestion
Add the artifacts under `build/install/xenocp/bin` to your `PATH`.
```

File: README.md [hunk 9]
```diff
@@ -159,25 +191,75 @@ and run the following command to produce output from the included sample data. T
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
 
 ## Seeking help
 
-[TODO]
+For questions and bug reports, please open an issue on the GitHub project page.
```
Comment (zaeleus) [inline README.md]:
```suggestion
For questions and bug reports, please open an issue on the [GitHub] project page.
```

File: README.md [hunk 10]
```diff
@@ -159,25 +191,75 @@ and run the following command to produce output from the included sample data. T
...
 ```
 
 [Dockerfile]: ./Dockerfile
 
+## Evaluate test data results
+
+If you have [bcftools] and a [GRCh37-lite] reference file, the following will show two variants in the input file. The variant on chromosome 1 is a variant in the host genome. The variant on chromosome 9 is a variant in the graft genome. 
+
+```
+$ bcftools mpileup  -R sample_data/output_data/regions.bed -f ref/GRCh37-lite/GRCh37-lite.fa sample_data/input_data/SJRB001_X.subset.bam | bcftools call -m - | tail -n 3
```
Comment (zaeleus) [inline README.md]:
```suggestion
$ bcftools mpileup -R sample_data/output_data/regions.bed -f ref/GRCh37-lite/GRCh37-lite.fa sample_data/input_data/SJRB001_X.subset.bam | bcftools call -m - | tail -n 3
```

File: README.md [hunk 11]
```diff
@@ -159,25 +191,75 @@ and run the following command to produce output from the included sample data. T
...
+#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	H_LC-SJRB001-X-SJ39.3-8L
+1	156044156	.	C	T	67	.	DP=49;VDB=0.525878;SGB=-0.670168;RPB=0.999118;MQB=0.00218214;MQSB=0.948436;BQB=0.743365;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=15,19,2,8;MQ=52	GT:PL	0/1:102,0,255
+9	19451994	.	G	A	182	.	DP=26;VDB=0.130558;SGB=-0.680642;RPB=0.887078;MQB=0.948139;MQSB=0.955682;BQB=0.053431;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=5,8,6,6;MQ=58	GT:PL	0/1:215,0,255
+```
+
+After running XenoCP, the host genome variant should be removed as the supporting reads will be unmapped. The following command demonstrates the removal of the variant on chromosome 1 in the output of the sample data.
+
+
+```
+$ bcftools mpileup  -R sample_data/output_data/regions.bed -f ref/GRCh37-lite/GRCh37-lite.fa sample_data/output_data/SJRB001_X.subset.xenocp.bam | bcftools call -m - | tail -n 3
```
Comment (zaeleus) [inline README.md]:
```suggestion
$ bcftools mpileup -R sample_data/output_data/regions.bed -f ref/GRCh37-lite/GRCh37-lite.fa sample_data/output_data/SJRB001_X.subset.xenocp.bam | bcftools call -m - | tail -n 3
```

File: README.md [hunk 12]
```diff
@@ -159,25 +191,75 @@ and run the following command to produce output from the included sample data. T
...
+
+Output: 
+```
+[mpileup] 1 samples in 1 input files
+#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	H_LC-SJRB001-X-SJ39.3-8L
+1	156044156	.	C	T	67	.	DP=49;VDB=0.525878;SGB=-0.670168;RPB=0.999118;MQB=0.00218214;MQSB=0.948436;BQB=0.743365;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=15,19,2,8;MQ=52	GT:PL	0/1:102,0,255
+9	19451994	.	G	A	182	.	DP=26;VDB=0.130558;SGB=-0.680642;RPB=0.887078;MQB=0.948139;MQSB=0.955682;BQB=0.053431;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=5,8,6,6;MQ=58	GT:PL	0/1:215,0,255
+```
+
+After running XenoCP, the host genome variant should be removed as the supporting reads will be unmapped. The following command demonstrates the removal of the variant on chromosome 1 in the output of the sample data.
```
Comment (zaeleus) [inline README.md]:
```suggestion
After running XenoCP, the host genome variant is removed, as the supporting reads will be unmapped. The following command demonstrates the removal of the variant on chromosome 1 in the output of the sample data.
```

File: README.md [hunk 13]
```diff
@@ -1,23 +1,39 @@
 # XenoCP
 
+XenoCP is a tool for cleansing mouse reads in xenograft BAMs.
+XenoCP can be easily incorporated into any workflow as it takes a BAM file
```
Comment (zaeleus) [inline README.md]:
```suggestion
XenoCP can be easily incorporated into any workflow, as it takes a BAM file
```

File: README.md [hunk 14]
```diff
@@ -1,23 +1,39 @@
...
+as input and efficiently cleans up the mouse contamination and gives a clean
```
Comment (zaeleus) [inline README.md]:
```suggestion
as input and efficiently cleans up the mouse contamination. The output is a clean
```

File: README.md [hunk 15]
```diff
@@ -1,23 +1,39 @@
...
+human BAM output that could be used for downstream genomic analysis.
```
Comment (zaeleus) [inline README.md]:
```suggestion
human BAM file that could be used for downstream genomic analysis. 
```

File: README.md [hunk 16]
```diff
@@ -55,30 +71,28 @@ disabled.
...
 [sambamba]: http://lomereiter.github.io/sambamba/
 
-## Building
 
-Once the prerequisites are satisfied, build XenoCP using Gradle. 
 
-```
-$ gradle :xenocp:installDist
-```
+## Local Usage
```
Comment (zaeleus) [inline README.md]:
```suggestion
## Local usage
```

File: README.md [hunk 17]
```diff
@@ -1,23 +1,39 @@
...
+	
+	# Test run on small dataset
 	mkdir results
-	cwltool --outdir results cwl/xenocp.cwl sample_data/input_data/inputs_local.yml
+	docker run \
+	  --mount type=bind,source=$(pwd)/sample_data/input_data,target=/data,readonly \
+	  --mount type=bind,source=$(pwd)/reference,target=/reference,readonly \
+	  --mount type=bind,source=$(pwd)/results,target=/results \
+	  xenocp \
+	  /data/inputs.yml
```
Comment (mcrusch) [inline README.md]: Did you confirm that these directions will work?

File: README.md [hunk 18]
```diff
@@ -1,23 +1,39 @@
...
+as input and efficiently cleans up the mouse contamination. The output is a clean
+human BAM file that could be used for downstream genomic analysis. 
+
 ## Getting started
 
-	git clone https://github.com/stjude/XenoCP.git
-	cd XenoCP
-	gradle :xenocp:installDist
+XenoCP can be run in the cloud on DNAnexus at
+https://platform.dnanexus.com/app/stjude_xenocp
```
Comment (mcrusch) [inline README.md]: Is there a risk people will have permissions issues with this?  I seem to remember something about the app being available to people with a St. Jude Cloud users group of some sort that they got automatically after logging into St. Jude Cloud Platform, or something along those lines.

File: README.md (hunk 6 above)
Comment (mcrusch) [reply README.md]: Original is better.  If it's unclear, perhaps: "copy of original BAM where contaminating host reads have been marked as unmapped"

File: README.md [hunk 19]
```diff
@@ -147,37 +179,87 @@ container, not the host, e.g.,
...
   --mount type=bind,source=$(pwd)/results,target=/results \
   xenocp \
   /data/inputs.yml
 ```
 
 [Dockerfile]: ./Dockerfile
 
+## Evaluate test data results
+
+If you have [bcftools] and a [GRCh37-lite] reference file, the following will show two variants in the input file. The variant on chromosome 1 is a variant in the host genome. The variant on chromosome 9 is a variant in the graft genome.
```
Comment (mcrusch) [inline README.md]: Should probably break across lines for source readability.

Also, I'm assuming the variant on chromosome 1 is not really a variant in the host genome, but probably a non-variant in the host genome that appears to be a variant when mapped against the graft genome.  So, perhaps say "The variant on chromosome 1 is an artifact of mouse reads.  The variant on chromosome 9..."

File: README.md (hunk 17 above)
Comment (adthrasher) [reply README.md]: Yes, I walked through the example and corrected the instructions as I went.

File: README.md (hunk 19 above)
Comment (adthrasher) [inline README.md]:
```suggestion
If you have [bcftools] and a [GRCh37-lite] reference file, the following will show two variants in the input file. 
The variant on chromosome 1 is an artifact of mouse reads. The variant on chromosome 9 is a variant in the graft genome. 
```

File: README.md (hunk 18 above)
Comment (adthrasher) [reply README.md]: @jsunny23 - Can you comment on this?

The dxapp config has the following set of users. 
`  "authorizedUsers": [
      "user-lding",
      "user-appdpdnanexus_stjude.org",
      "user-appdddnanexus_stjude.org",
      "org-stjude_cloud_app_runners",
      "org-stjude_cloud"
  ],`

Comment (jsunny23) [reply README.md]: Yes, when a user logs into St. Jude Cloud Platform, they're added to `org-stjude_cloud_app_runners`. So all St. Jude Cloud users should be able to see it under the tools page (I can see the app) and be vendable from platform. 

Note: DNAnexus only allow developers or authorized users to run apps. So if you want this to be truly public, that may require talking to DNAnexus.

Comment (claymcleod) [reply README.md]: I think you should probably specify that you'll need to login to St. Jude Cloud at least once to get access to the app.

Comment (zaeleus) [reply README.md]: For reference, when you try to make an app `PUBLIC` on DNAnexus:

```
***
WARNING: app-mtsg will be made public. Anyone will be able to view and run all
published versions of this app.
***
Confirm making this app public [y/N]: y
Error: You are not authorized to add "PUBLIC" to the authorized users list for app mtsg; if you wish to make an app or a workflow available to all users, then please contact support, code 401.
```

Comment (mcrusch) [reply README.md]: I have contacted DNAnexus support.  Their response led me to believe this is going to be a slow process.  In the meantime, I have added instructions on how to login to St. Jude Cloud to get access to the app at the permalink page for the paper:

https://pecan.stjude.cloud/permalink/xenocp

Could somebody from the cloud team make sure the instructions I wrote are correct?  If so, we can update the README to point to that permalink page.  Then, once it's truly public, we can go back and update again.

**Comment by zaeleus** [comment]
The last two commits (e4b0aab and 87ea227) are irrelevant to this PR. These should have been on a new branch.

**Pull request merged**